### DISCUSSION TITLE
Confused about visitor access and albums

### DISCUSSION THREAD

**Discussion by bythelake9428** [photoprism/photoprism#3386] [General]
I'm truly confused about the multi-user features and may need to re-think my plans for Photoprism.

I'm a Photoprism Plus user today, and I'd like to make my Photoprism instance accessible (read-only) to the public, **without requiring a log-in.**

I see that there are both Guest and Visitor roles, but the descriptions state that each can view resources "that have been shared with them".  I have no albums defined today, and didn't intend on creating albums, as I want my whole photo archive to be accessible to the public.

My question is this:  Is there a way for me to share my Photoprism instance publically, without requiring authentication, where  visitors can search my entire photo collection instead of a specific album?

**Comment by graciousgrey** [comment]
This sounds very similar to this feature request (which is not yet implemented)

- https://github.com/photoprism/photoprism/issues/669

At the moment you could use  either the viewer role (however this requires a login)

https://docs.photoprism.app/user-guide/users/#viewer

or album sharing links 

https://docs.photoprism.app/user-guide/share/

**Reply by bythelake9428** [reply]
Thanks for taking the time to reply.  Truly appreciated.

Seems like true public, anonymous access isn't happening anytime soon.  Visitor access requires a login, and the feature request you pointed out from 2020 hasn't been assigned.  I already have an online photo archive, but I wanted to expose Photoprism for its search capability.

I also simply wanted to share all 37,000 photos via the search.  It seems that Photoprism's sharing capability is via an Album, which would require that I add all 37,000 photos to an Album, then add new pictures to the album every time I add photos to my archive.  As the admin, I can search the whole collection without having an album defined, and I was hoping to share the same with the public.

I may need to re-think my public photo search strategy.
